include/envoy/thread/thread.h

```diff
@@ -21,6 +22,20 @@ class Thread {
 
 typedef std::unique_ptr<Thread> ThreadPtr;
 
+/**
+ * Interface providing a mechanism for creating threads.
+ */
+class ThreadSystem {
```

Comment:
total bikeshed: I would probably call this `ThreadFactory` as that is generally the naming pattern we do in the code. TIOLI.

Reply:
done